core/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala

```diff
@@ -144,6 +145,19 @@ abstract class AppBase(
     }
   }
 
+  def getSparkEventFromJson(): (String) => org.apache.spark.scheduler.SparkListenerEvent = {
+    // Spark 3.4 and Databricks changed the signature on sparkEventFromJson
+    val c = Class.forName("org.apache.spark.util.JsonProtocol")
+    try {
+      val m = c.getDeclaredMethod("sparkEventFromJson", classOf[org.json4s.JValue])
+      (line: String) => m.invoke(null, parse(line)).asInstanceOf[org.apache.spark.scheduler.SparkListenerEvent]
+    } catch {
+      case NonFatal(_) =>
+        val m = c.getDeclaredMethod("sparkEventFromJson", classOf[String])
+        (line: String) => m.invoke(null, line).asInstanceOf[org.apache.spark.scheduler.SparkListenerEvent]
+    }
```

Comment:
Scala's `Try` can make this more succinct.
```suggestion
    val c = Class.forName("org.apache.spark.util.JsonProtocol")
    Try {
      val m = c.getDeclaredMethod("sparkEventFromJson", classOf[org.json4s.JValue])
      (line: String) => m.invoke(null, parse(line)).asInstanceOf[org.apache.spark.scheduler.SparkListenerEvent]
    }.getOrElse {
      val m = c.getDeclaredMethod("sparkEventFromJson", classOf[String])
      (line: String) => m.invoke(null, line).asInstanceOf[org.apache.spark.scheduler.SparkListenerEvent]
    }
```

Reply:
I'll update myself so I can fix up imports